dotnet/src/SemanticKernel/Planning/FunctionFlowRunner.cs

```diff
@@ -154,12 +168,56 @@ private async Task<string> ProcessNodeListAsync(XmlNodeList nodeList, string fun
                     continue;
                 }
 
+                // Ignoring first ELSE after the If that was already processed
+                if (ignoreElse && o2.Name.StartsWith(ConditionElseTag, StringComparison.InvariantCultureIgnoreCase))
+                {
+                    context.Log.LogTrace("{0}: ignoring ELSE tag node", parentNodeName);
+                    ignoreElse = false;
+                    continue;
+                }
+
+                if (processFunctions && o2.Name.StartsWith(ConditionIfTag, StringComparison.InvariantCultureIgnoreCase))
+                {
+                    context.Log.LogTrace("{0}: found IF tag node", parentNodeName);
+                    // Includes IF + ELSE statement
+                    var ifFullContent = o2.OuterXml;
+
+                    //Go for the next node to see if it's an else
+                    if (this.CheckIfNextNodeIsElseAndGetItsContents(o2, out var elseContents))
+                    {
+                        ifFullContent += elseContents;
+
+                        // Ignore the next else sibling tag to avoid adding recursively it to the next plan iteration.
+                        ignoreElse = true;
+                    }
+
+                    var functionVariables = new ContextVariables();
+                    functionVariables.Update(context.Variables);
+                    functionVariables.Set("INPUT", ifFullContent);
+
+                    var branchIfOrElse = await this._conditionalFlowHelper.IfAsync(ifFullContent,
+                        new SKContext(functionVariables, this._kernel.Memory, this._kernel.Skills, this._kernel.Log,
+                            context.CancellationToken));
+
+                    _ = stepAndTextResults.Append(INDENT).AppendLine(branchIfOrElse);
+
+                    processFunctions = false;
+                    
+                    // We need to continue so we don't ignore any next siblings (If or Function) 
+                    continue;
+                }
+
                 if (o2.Name.StartsWith(FunctionTag, StringComparison.InvariantCultureIgnoreCase))
                 {
                     var skillFunctionName = o2.Name.Split(FunctionTag)?[1] ?? string.Empty;
                     context.Log.LogTrace("{0}: found skill node {1}", parentNodeName, skillFunctionName);
                     GetSkillFunctionNames(skillFunctionName, out var skillName, out var functionName);
-                    if (processFunctions && !string.IsNullOrEmpty(functionName) && context.IsFunctionRegistered(skillName, functionName, out var skillFunction))
+                    if (!context.IsFunctionRegistered(skillName, functionName, out var skillFunction))
```

Comment:
Maybe this if/throw should be inside the if(processFunctions) conditional?

Reply:
I moved this outside because plan was ignoring a non registered functions (from existing skills).

The IsFunctionRegistered doens't throw when the function is not found but the skill exists. 

This change stops the Plan if it attempts to use a function that doesn't exists.

Reply:
I added UnitTests to cover all those scenarios as well.